core/process/last_status_spec.rb

```diff
@@ -0,0 +1,19 @@
+require File.expand_path('../../../spec_helper', __FILE__)
+
+ruby_version_is '2.5' do
+  describe 'Process#last_status' do
+    it 'returns the status of the last executed child process in the current thread' do
+      pid = Process.spawn("ruby", "-e", "exit 13")
+      Process.wait pid
+      Process.last_status.pid.should == pid
+    end
+
+    it 'returns nil if no child process has been ever executed in the current thread' do
+      Process.last_status.should == nil
```

Comment:
This should be executed in a new thread so it's guaranteed there was no subprocess before.

Reply:
How about if we run it before the spec 1?. I have updated the PR to run this nil check spec first.

Reply:
`Process.last_status` like `$?` is thread-local I believe.
So if a previous spec used a subprocess (very likely) then it should fail.

Reply:
I am new to using the Process class. Regarding -- `executed in a new thread` I tried --
``` 
th = Thread.new do
   Process.last_status.should == nil
end
```
which outputs --
```
Expected #<Process::Status: pid 30790 exit 0>
 to equal nil```
How should we execute it in a new thread ?

Reply:
You can do something like:
```ruby
it "returns nil if no child process has been ever executed in the current thread" do
  Thread.new {
    Process.last_status.should == nil
  }.join
end
```
So the problem is the Thread was not `join`-ed which means the expectation might not have executed before the spec ends.

Reply:
Yes, that works. I have updated the PR.

Reply:
Maybe activating the leak checker by default (`CHECK_LEAKS=t`) would help to find what goes wrong in such a case. I'll think about it.